apps/src/lib/kits/maker/boards/microBit/MBFirmataWrapper.js

```diff
@@ -9,14 +10,60 @@ export default class MicrobitFirmataWrapper extends MBFirmataClient {
     this.digitalCallbacks = [];
   }
 
-  connectBoard() {
+  connectBoard(port) {
     return Promise.resolve()
-      .then(() => this.connect())
+      .then(() => this.setSerialPort(port))
       .then(() => {
         return this.setAnalogSamplingInterval(SAMPLE_INTERVAL);
       });
   }
 
+  // Used in setSerialPort
+  dataReceived(data) {
+    if (this.inbufCount + data.length < this.inbuf.length) {
+      this.inbuf.set(data, this.inbufCount);
+      this.inbufCount += data.length;
+      this.processFirmatMessages();
+    }
+  }
+
+  setSerialPort(port) {
+    if (isNodeSerialAvailable()) {
+      return super.setSerialPort(port);
+    } else {
+      // Use the given port. Assume the port has been opened by the caller.
+
+      this.myPort = port;
+      this.myPort.on('data', this.dataReceived.bind(this));
+      this.requestFirmataVersion();
+      this.requestFirmwareVersion();
+
+      // get the board serial number (used to determine board version)
+      this.boardVersion = '';
+      return Promise.resolve();
+
+      // The below code snippet is from the MBFirmataClient setSerialPort
+      // that is not included in this branch for the ChromeSerialPort.
+      // The Chrome Serial Port doesn't return .list() as a promise, so this\
+      // code doesn't work in that library. As of 11/11/20, we do not use the
+      // boardVersion in our MB integration.
+      // TODO - rewrite this promise code to work for the Chrome Serial Port.
```

Comment:
Are we missing some functionality until this rewrite happens?

Reply:
Not as far as I know. It looks like this is what tells version 1.3 and version 1.5 of the boards apart, but we don't differ behavior for those boards, so it shouldn't matter to our current behavior.

Reply:
In that case, I think it might be worth removing this code and documenting it for future possible use elsewhere?  (Maybe just in a comment in this PR?)  I think we should only leave commented-out code in the source tree in special cases, and if it doesn't seem like this is really an imminent TODO, then it might not quite qualify.

Reply:
I definitely agree that it's not the neatest thing to leave here. One of the reasons I originally left it there was to emphasize the difference between each branch - on line 32 we go to the super.setSerialPort, and here we do almost the same thing as the super.setSerialPort EXCEPT those lines that are commented out. If the comments about the commented lines of code were clearer, do you think it would make sense to leave it or should it still be documented elsewhere?

Reply:
If you feel that it will make things clearer for future visitors to the code, then it seems okay to leave it in, but I would make clear in the comments why it's been left here and that it's really more of a comment than pending code, and remove the `"TODO"` which implies that there is some unfinished work here.